modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/Instruments.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.apm.internal.metrics;
+
+import io.opentelemetry.api.metrics.Meter;
+
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.common.util.concurrent.ReleasableLock;
+import org.elasticsearch.telemetry.MetricName;
+import org.elasticsearch.telemetry.metric.DoubleCounter;
+import org.elasticsearch.telemetry.metric.DoubleGauge;
+import org.elasticsearch.telemetry.metric.DoubleHistogram;
+import org.elasticsearch.telemetry.metric.DoubleUpDownCounter;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongGauge;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.LongUpDownCounter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class Instruments {
+    private final Registrar<DoubleCounterAdapter<?>> doubleCounters = new Registrar<>();
+    private final Registrar<DoubleUpDownCounterAdapter<?>> doubleUpDownCounters = new Registrar<>();
+    private final Registrar<DoubleGaugeAdapter<?>> doubleGauges = new Registrar<>();
+    private final Registrar<DoubleHistogramAdapter<?>> doubleHistograms = new Registrar<>();
+    private final Registrar<LongCounterAdapter<?>> longCounters = new Registrar<>();
+    private final Registrar<LongUpDownCounterAdapter<?>> longUpDownCounters = new Registrar<>();
+    private final Registrar<LongGaugeAdapter<?>> longGauges = new Registrar<>();
+    private final Registrar<LongHistogramAdapter<?>> longHistograms = new Registrar<>();
+
+    private final List<Registrar<?>> registrars = List.of(
+        doubleCounters,
+        doubleUpDownCounters,
+        doubleGauges,
+        doubleHistograms,
+        longCounters,
+        longUpDownCounters,
+        longGauges,
+        longHistograms
+    );
+
+    protected final ReleasableLock registerLock = new ReleasableLock(new ReentrantLock());
```

Comment:
do we really need to lock?
from just reading the code it looks like it should be safe without it?